sherpa/plot/backends.py

```diff
@@ -0,0 +1,961 @@
+              'overplot': ['bool',
+                           'If `True`, the plot is added to an existing plot, if not a new plot is created.'],
+              'clearwindow': ['bool',
+                              'If `True` the entire figure area is cleared to make space for a new plot.'],
+              'xerrorbars': ['bool',
+                             '''Should x error bars be shown? If this is set to `True` errorbars
+are shown, but only if the size of the errorbars is provided in the
+`xerr` parameters. The purpose of having a separate switch
+`xerrorbars` is that the prepare method of a plot can create the
+errors and pass them to this method, but the user can still decide
+to change the style of the plot and choose if error bars should be
+displayed.'''],
+              'yerrorbars': ['bool',
+                             '''Should y error bars be shown? If this is set to `True` errorbars
+are shown, but only if the size of the errorbars is provided in the
+`yerr` parameters. The purpose of having a separate switch
+`yerrorbars` is that the prepare method of a plot can create the
+errors and pass them to this method, but the user can still decide
+to change the style of the plot and choose if error bars should be
+displayed.'''],
+              'color': ['string',
+                        '''The following colors are accepted by all backends: ``'b'`` (blue),
+``'r'`` (red), ``'g'`` (green), ``'k'`` (black), ``'w'`` (white),
+``'c'`` (cyan), ``'y'`` (yellow), ``'m'`` (magenta) but they may not
+translate to the exact same RGB values in each backend, e.g. ``'b'``
+could be a different shade of blue depending on the backend.
+Some backends might accept additional values.'''],
+              'linestyle': ['string',
+                            '''The following values are accepted by all backends: ``'noline'``,
+``'None'`` (as string, same as ``'noline'``),
+``'solid'``, ``'dot'``, ``'dash'``, ``'dotdash'``, ``'-'`` (solid
+line), ``':'`` (dotted), ``'--'`` (dashed), ``'-.'`` (dot-dashed),
+``''`` (empty string, no line shown), `None` (default - usually
+solid line).
+Some backends may accept additional values.'''],
+              'linewidth': ['float', 'Thickness of the line.'],
+              'linecolor': ['string',
+                            '''The following colors are accepted by all backends: ``'b'`` (blue),
+``'r'`` (red), ``'g'`` (green), ``'k'`` (black), ``'w'`` (white),
+``'c'`` (cyan), ``'y'`` (yellow), ``'m'`` (magenta) but they may not
+translate to the exact same RGB values in each backend, e.g. ``'b'``
+could be a different shade of blue depending on the backend.
+Some backend might accept additional values.'''],
+              'marker': ['string',
+                         '''The following values are accepted by all backends: "None" (as a
+string, no marker shown), "." (dot), "o" (circle), "+", "s" (square), "" (empty string, no marker shown)
```

Comment:
Let's default to a circle.

Reply:
But in CIAO 4.15, `None` means "no marker". That's not consistent with what we do in other cases, but it is what is is. I guess the reason for that is that the default for a plot is "solid line with no dots in it" and you get that default with `linestyle=None, marker=None`. So, I'll fix the comment, but won't change the API - too many exiting plot scripts would break to do that without through discussion.